File: javascript/invoice_item.js
class InvoiceItem {
  constructor(spendType, amount, transaction) {
    this.spendType = spendType;
    this.amount = amount;
    this.transaction = transaction;
  }

  toString() {
    return `InvoiceItem for $${this.amount}: ${this.spendType} for ${this.transaction.name}`;
  }
}

export default InvoiceItem;

File: dotnet/MiniThatch/InvoiceItem.cs
namespace MiniThatch;
public class InvoiceItem
{
    public string SpendType { get; set; }
    public double Amount { get; set; }
    public Transaction Transaction { get; set; }

    public InvoiceItem(string spendType, double amount, Transaction transaction)
    {
        SpendType = spendType;
        Amount = amount;
        Transaction = transaction;
    }

    public override string ToString()
    {
        return $"InvoiceItem for ${Amount}: {SpendType} for {Transaction.Name}";
    }
}
